### PR TITLE
Add script that generates charities.json.

### PR DESCRIPTION
**Pull request by ryan-draper**

### CODE REVIEW

File: charity-data/getcharities.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+sort = 'RATING'
+params = {'app_id': app_id, 'app_key': app_key, 'pageSize': pageSize,
+          'rated': rated, 'sizeRange': sizeRange, 'sort': sort}
+
+url = base_url + parse.urlencode(params)
+
+# Get JSON response body
+charity_nav_json = requests.get(url).json()
+
+# Counters for the number of charities with no valid donate link, the number
```
Comment (tonioshikanlu) [inline charity-data/getcharities.py:22]: This code could be split up into functions.

Comment (charu0291) [reply charity-data/getcharities.py:22]: +1

File: charity-data/getcharities.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+num_no_valid_donate = 0
+num_no_valid_link = 0
+num_no_image = 0
+
+charities = []
+for charity in charity_nav_json:
+    # Create a charity JSON object
+    processed_charity = {}
+    
+    # Name
```
Comment (charu0291) [inline charity-data/getcharities.py:33]: you can remove such comments which are trivial and not explain much anyways

File: charity-data/getcharities.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    processed_charity['name'] = charity['charityName']
+
+    # Link
```
Comment (charu0291) [inline charity-data/getcharities.py:36]: remove comment

File: charity-data/getcharities.py (hunk 1 above)
Comment (ryan-draper) [reply charity-data/getcharities.py:37]: I created a helper function for getting the charity's link. I considered making one for getting the image link, but that's only an if statement, so modularization of that seemed excessive.

File: charity-data/getcharities.py (hunk 3 above)
Comment (ryan-draper) [reply charity-data/getcharities.py]: Done.

File: charity-data/getcharities.py (hunk 2 above)
Comment (ryan-draper) [reply charity-data/getcharities.py]: Done.

File: charity-data/getcharities.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import requests
+import json
+
+
+def get_charity_link(base_charity_url):
+    # Append / if not already present
+    if base_charity_url[len(base_charity_url) - 1] != '/':
+        base_charity_url += '/'
+    # If /donate is valid
+    if requests.get(base_charity_url + 'donate').status_code < 400:
```
Comment (charu0291) [inline charity-data/getcharities.py:11]: should we check for exact status code, like 200, 302 etc instead of generic condition.

Comment (ryan-draper) [reply charity-data/getcharities.py:11]: We could check for specific conditions but it'd be difficult to know which status codes to look for. Any code under 400 indicates a valid link (Informational responses (100–199), Successful responses (200–299), Redirects (300–399)) according to [https://developer.mozilla.org/en-US/docs/Web/HTTP/Status](url).

Comment (curves4d) [reply charity-data/getcharities.py:11]: You should check for Status code 200 OK. the success responses are in the ange 200-299 and 200 is the only relevant one here.

File: charity-data/getcharities.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            num_no_valid_link += 1
+    except requests.ConnectionError as e:
+        print("Failed to open url " + charity['websiteURL'])
+        num_no_valid_link += 1
+        continue
+
+    # NOTE: WE HAVE TO ATTRIBUTE CLEARBIT API:
+    # <a href="https://clearbit.com">Logos provided by Clearbit</a>
+    try:
+        if requests.get('https://logo.clearbit.com/' + charity['websiteURL']).status_code < 400:
```
Comment (curves4d) [inline charity-data/getcharities.py:63]: same here. check status code 200

File: charity-data/getcharities.py (hunk 1 above)
Comment (curves4d) [reply charity-data/getcharities.py:37]: I think what's intended here is that you code should be modularized into testable and reusable blocks and not a bunch of statement at the top level. Please break the logic into  independent functions. eg. get_charities_json(), process_charity(charity), getCharityLink(charity), getCharityImage(charity)

Also please use the if __name__ == "__main__" construct in your scripts. https://www.freecodecamp.org/news/if-name-main-python-example/#:~:text=We%20can%20use%20an%20if,name%20if%20it%20is%20imported.

Comment (ryan-draper) [reply charity-data/getcharities.py]: Done.

File: charity-data/getcharities.py (hunk 4 above)
Comment (ryan-draper) [reply charity-data/getcharities.py]: Done.

File: charity-data/getcharities.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    except requests.ConnectionError as e:
+        print("Failed to open url " + 'https://logo.clearbit.com/' + charity_url)
+        num_no_image += 1
+        continue
+
+    # Category
+    processed_charity['category'] = charity['category']['categoryName']
+    # Description
+    processed_charity['description'] = charity['mission']
+    # Rating
```
[the remark this answers is not in the document]
Comment (ryan-draper) [reply charity-data/getcharities.py]: Done.

File: charity-data/getcharities.py (hunk 5 above)
Comment (ryan-draper) [reply charity-data/getcharities.py]: Done.

**Pull request merged**